### PR TITLE
MAINT: Printing Error Msg to STDERR and Removing iterate.dat

### PR DESCRIPTION
**Pull request by sumitbinnani**
Printing Error Msg to STDERR in place of STDOUT
Removed the `iterate.dat` from LBFGS
See #5037

### CODE REVIEW

**Comment by larsmans** [comment]
`scipy/optimize/lbfgsb.py` should be updated as well.

**Comment by sumitbinnani** [comment]
@larsmans Sorry wasn't in town. Would make the required changes asap.

**Comment by sumitbinnani** [comment]
@larsmans Not able to grasp the changes to be made in ```scipy/optimize/lbfgsb.py```. Kindly guide.

**Comment by larsmans** [comment]
The docstring for `fmin_l_bfgs_b` still describes the old logging behavior.

**Comment by sumitbinnani** [comment]
Thanks @larsmans.
Made the change in documentation of ```fmin_l_bfgs_b``` to account for new logging behavior.

**Comment by sumitbinnani** [comment]
@larsmans @ev-br
Does the changes made in PR meet the requirements? Kindly let me know if any additional changes are required.

**Comment by larsmans** [comment]
LGTM, but I'll let someone who can actually read Fortran do the merging :)

**Comment by ev-br** [comment]
I read Fortran, and the code seems OK. But the docstring is not very consistent: it only mentions `iprint < 0` and `iprint==0`, while the code recognizes positive values of `iprint`. To check, try with
```
In 10]: from scipy.optimize import fmin_l_bfgs_b as lbfgs

In [11]: def f(x): return (x - 2.)**2, 2.*(x - 2.)

In [12]: lbfgs(f, 1., iprint=101)
 
```
and change iprint to 1 or 99. I'm not sure many people use this (partly because I don't think many people need this or are aware of this); still I'd rather avoid giving misleading information. Ideally, the Fortran level description of the iprint variable (e.g., https://github.com/scipy/scipy/pull/5340/files#diff-b68fd6e59113bfe219e7099876e38b34L130) makes its way to the python level docstring.

**Comment by sumitbinnani** [comment]
Have updated python docs to reflect fortran code. Is it okay, @ev-br?

**Comment by ev-br** [comment]
Right. I *think* it's because `|` symbol has a special meaning in reST, so it needs to be shielded by double backicks, like so ``|stuff|``.
You can check the doc build locally by `python runtests.py --refguide-check``

File: scipy/optimize/lbfgsb.py [hunk 1]
```diff
@@ -95,9 +95,11 @@ def fmin_l_bfgs_b(func, x0, fprime=None, args=(),
         calculating the gradient
     iprint : int, optional
         Controls the frequency of output. ``iprint < 0`` means no output;
-        ``iprint == 0`` means write messages to stdout; ``iprint > 1`` in
-        addition means write logging information to a file named
-        ``iterate.dat`` in the current working directory.
+        ``iprint = 0``    print only one line at the last iteration;
+        ``0 < iprint < 99`` print also f and |proj g| every iprint iterations;
```
Comment (ev-br) [inline scipy/optimize/lbfgsb.py]: Here, |proj g| needs to be wrapped into double backicks.

**Comment by sumitbinnani** [comment]
Wrapping |proj g| into double backicks solved the reST issue. Thanks @ev-br :-)

Let me know if any additional changes are required.

**Comment by ev-br** [comment]
Great! Merging, thank you @sumitbinnani and congratulations with what I believe is your first SciPy PR. Keep them coming!

**Comment by larsmans** [comment]
Thanks @sumitbinnani!

**Comment by sumitbinnani** [comment]
Happy to make my first contribution to SciPy :-)
Thanks @ev-br and @larsmans :-)

**Pull request merged**